com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs

```diff
@@ -9,61 +9,29 @@ namespace Unity.MLAgents.Editor
 {
     internal class MLAgentsSettingsBuildProvider : IPreprocessBuildWithReport, IPostprocessBuildWithReport
     {
-        private MLAgentsSettings m_SettingsAddedToPreloadedAssets;
-
         public int callbackOrder => 0;
 
         public void OnPreprocessBuild(BuildReport report)
         {
-            var wasDirty = IsPlayerSettingsDirty();
-            m_SettingsAddedToPreloadedAssets = null;
+            if (!EditorUtility.IsPersistent(MLAgentsSettingsManager.Settings))
+                return;
 
             var preloadedAssets = PlayerSettings.GetPreloadedAssets().ToList();
             if (!preloadedAssets.Contains(MLAgentsSettingsManager.Settings))
             {
-                m_SettingsAddedToPreloadedAssets = MLAgentsSettingsManager.Settings;
-                preloadedAssets.Add(m_SettingsAddedToPreloadedAssets);
+                preloadedAssets.Add(MLAgentsSettingsManager.Settings);
```

Comment:
is `MLAgentsSettingsManager.Settings` guaranteed to be non-null? I guess so, I found this in a test file:

`Assert.AreNotEqual(null, MLAgentsSettingsManager.Settings);`

but not 100% sure

Reply:
Yes. See https://github.com/Unity-Technologies/ml-agents/blob/6bb711f1b0a29a06b4deb544bb1e93c392acb255/com.unity.ml-agents/Runtime/MLAgentsSettingsManager.cs#L22.